extensions/amp-access/0.1/amp-access.js

```diff
@@ -277,14 +280,24 @@ export class AccessService {
 
   /**
    * @param {string} url
+   * @param {boolean} useAuthData Allows `AUTH(field)` URL var substitutions.
    * @return {!Promise<string>}
    * @private
    */
-  buildUrl_(url) {
+  buildUrl_(url, useAuthData) {
     return this.getReaderId_().then(readerId => {
-      return this.urlReplacements_.expand(url, {
+      const vars = {
         'READER_ID': readerId
-      });
+      };
+      if (useAuthData) {
+        vars['AUTH'] = field => {
+          if (this.authResponse_) {
+            return this.authResponse_[field];
```

Comment:
This may return `undefined`, which is different than the `null` we return without `authResponse_`. Should check if `field in authResponse_ ? authResponse_[field] : null`?